src/mtcp/mtcp_restart.c

```diff
@@ -235,12 +238,18 @@ main(int argc, char *argv[], char **environ)
 #endif /* ifdef ENABLE_VDSO_CHECK */
 
   rinfo.fd = -1;
+  rinfo.mtcp_restart_pause = 0; /* false */
   rinfo.use_gdb = 0;
   rinfo.text_offset = -1;
   shift;
   while (argc > 0) {
+    // All parsing of flags with no arg must appear before parsing ckptImage
+    if (mtcp_strcmp(argv[0], "--mtcp-restart-pause") == 0) {
```

Comment:
I see, our current logic is faulty. The `argc == 1` test should be moved near the last `else` statement. Otherwise, we will see the same error with `mtcp_restart <args> --simulate`, `mtcp_restart <args> --use-gdb`, and so on.